src/main/java/com/aws/greengrass/clientdevices/auth/session/SessionConfig.java

```diff
@@ -75,10 +100,33 @@ private int getConfiguredSessionCapacity() {
         int configValue = Coerce.toInt(configuration.findOrDefault(DEFAULT_SESSION_CAPACITY,
                 PERFORMANCE_TOPIC, MAX_ACTIVE_AUTH_TOKENS_TOPIC));
 
-        int clamped = Math.max(MIN_SESSION_CAPACITY, Math.min(MAX_SESSION_CAPACITY, configValue));
+        return getClampedConfigValue(MAX_ACTIVE_AUTH_TOKENS_TOPIC, configValue,
+                MIN_SESSION_CAPACITY, MAX_SESSION_CAPACITY);
+    }
+
+    /**
+     * Retrieves configured client device trust duration.
+     * Invalid values are clamped to the valid range
+     *
+     * @return session capacity value
+     */
+    private int getConfiguredTrustDuration() {
+        if (configuration == null || configuration.isEmpty()) {
+            return DEFAULT_CLIENT_DEVICE_TRUST_DURATION_HOURS;
+        }
+
+        int configValue = Coerce.toInt(configuration.findOrDefault(DEFAULT_CLIENT_DEVICE_TRUST_DURATION_HOURS,
+                SECURITY_TOPIC, CLIENT_DEVICE_TRUST_DURATION_HOURS_TOPIC));
+
+        return getClampedConfigValue(CLIENT_DEVICE_TRUST_DURATION_HOURS_TOPIC, configValue,
+                MIN_CLIENT_DEVICE_TRUST_DURATION_HOURS, MAX_CLIENT_DEVICE_TRUST_DURATION_HOURS);
+    }
+    
+    private int getClampedConfigValue(String configKey, int configValue, int min, int max) {
```

Comment:
Should we do this things for them or fail? - Specially for the TTL I think it might be better to break than just showing a log and tell them all is good.